src/anemoi/training/losses/mse.py

```diff
@@ -23,69 +24,87 @@ class WeightedMSELoss(nn.Module):
         self,
         node_weights: torch.Tensor,
-        data_variances: torch.Tensor | None = None,
-        ignore_nans: bool | None = False,
+        feature_weights: torch.Tensor | None = None,
+        ignore_nans: bool = False,
     ) -> None:
         """Latitude- and (inverse-)variance-weighted MSE Loss.
 
         Parameters
         ----------
-        node_weights : torch.Tensor of shape (N, )
-            Weight of each node in the loss function
-        data_variances : Optional[torch.Tensor], optional
+        node_weights : torch.Tensor
+            Weights by area
+        feature_weights : Optional[torch.Tensor], optional
             precomputed, per-variable stepwise variance estimate, by default None
         ignore_nans : bool, optional
             Allow nans in the loss and apply methods ignoring nans for measuring the loss, by default False
-
         """
         super().__init__()
 
         self.avg_function = torch.nanmean if ignore_nans else torch.mean
         self.sum_function = torch.nansum if ignore_nans else torch.sum
 
-        self.register_buffer("weights", node_weights, persistent=True)
-        if data_variances is not None:
-            self.register_buffer("ivar", data_variances, persistent=True)
+        self.register_buffer("node_weights", node_weights, persistent=True)
+        if feature_weights is not None:
+            self.register_buffer("feature_weights", feature_weights, persistent=True)
 
     def forward(
         self,
         pred: torch.Tensor,
         target: torch.Tensor,
         squash: bool = True,
+        feature_indices: torch.Tensor | None = None,
+        feature_scale: bool = True,
     ) -> torch.Tensor:
         """Calculates the lat-weighted MSE loss.
 
         Parameters
         ----------
         pred : torch.Tensor
-            Prediction tensor, shape (bs, lat*lon, n_outputs)
+            Prediction tensor, shape (bs, (optional_ensemble), lat*lon, n_outputs)
         target : torch.Tensor
-            Target tensor, shape (bs, lat*lon, n_outputs)
+            Target tensor, shape (bs, (optional_ensemble), lat*lon, n_outputs)
         squash : bool, optional
             Average last dimension, by default True
+        feature_indices:
+            feature indices (relative to full model output) of the features passed in pred and target
+        feature_scale:
+            If True, scale the loss by the feature_weights
 
         Returns
         -------
         torch.Tensor
             Weighted MSE loss
-
         """
+        if pred.ndim == 4:
+            pred = pred.mean(dim=1)
+
+        torch.save(self.node_weights, "node_weights.pt")
+        torch.save(pred, "pred.pt")
+        torch.save(target, "target.pt")
+
         out = torch.square(pred - target)
 
-        # Use variances if available
-        if hasattr(self, "ivar"):
-            out *= self.ivar
+        if feature_scale and hasattr(self, "feature_weights"):
+            out = (
```

Comment:
What's the performance implication of this change?

The operation before was in-place, this changes the space and time complexity.

Reply:
Reverted in next commit back to in-place